src/toast/pipeline_tools/classes.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2019 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import os
+import pickle
+import sys
+
+import numpy as np
+
+from ..timing import function_timer, Timer
+from ..tod import AnalyticNoise
+from ..utils import Logger, Environment
+from .. import qarray
+
+
+def name2id(name, maxval=2 ** 16):
+    """ Map a name into an index.
+    """
+    value = 0
+    for c in name:
+        value += ord(c)
+    return value % maxval
+
+
+class Focalplane:
+    detector_data = {}
+    _detweights = None
+    sample_rate = None
+    _radius = None
+    _detquats = None
+    _noise = None
```

Comment:
not sure why there are class attributes, they are generally used only in very special circumstances.

Reply:
Removed.